### PR TITLE
Optimization Workbook - Release Version - 1.0

### PR DESCRIPTION
**Pull request by arthurclares**
## 🛠️ Description
Added the latest version of the Optimization Workbook. This workbook now includes:
-Improve AHUB queries
-Summarized all Reservations opportunities in one single tab
-Added queries to check for Idle disks Snapshots and Old disks Snapshots (+30 days)

## 🔬 How has this been tested?
- [ ] 🫰 PS -WhatIf / az validate
- [x] 👍 Manually deployed + verified
- [x] 💪 Unit tests

### CODE REVIEW

File: docs/optimization-workbook/workbook/azuredeploy.json
Comment (flanakin) [inline docs/optimization-workbook/workbook/azuredeploy.json:1]: Did you mean to add this? This should be covered by the Bicep code, right?

**Review by flanakin — CHANGES_REQUESTED** [review]

**Comment by flanakin** [comment]
@arthurclares FYI - Remove the `Needs: Attention` label when you resolve the issues and need another review.

File: docs/optimization-workbook/workbook/azuredeploy.json
Comment (arthurclares) [reply docs/optimization-workbook/workbook/azuredeploy.json:1]: The idea was to use this while we don't have the Bicep code ready yet. Are you working to develop this code? I can give it a try as well.

File: docs/optimization-workbook/workbook/Optimization Workbook.workbook
[the remark this answers is not in the document]
Comment (arthurclares) [reply docs/optimization-workbook/workbook/Optimization Workbook.workbook:1]: Both .json and .workbook should work.

File: docs/optimization-workbook/workbook/placeholder
[the remark this answers is not in the document]
Comment (arthurclares) [reply docs/optimization-workbook/workbook/placeholder:1]: Github was acting weird yesterday, we don't need this.

File: docs/optimization-workbook/workbook/azuredeploy.json
Comment (flanakin) [reply docs/optimization-workbook/workbook/azuredeploy.json:1]: Bicep code is ready :-) Go to the src/scripts directory and run `./Deploy-Toolkit optimization -Build` (I think).

File: docs/optimization-workbook/workbook/Optimization Workbook.workbook
[the remark this answers is not in the document]
Comment (flanakin) [reply docs/optimization-workbook/workbook/Optimization Workbook.workbook:1]: The Build script requires the file to be `workbook.json`. Run `Build-Workbook optimization` to test the build script. it generates the files in the `release` folder.

File: docs/optimization-workbook/workbook/azuredeploy.json
Comment (arthurclares) [reply docs/optimization-workbook/workbook/azuredeploy.json:1]: Thank your for that, you are right, I have not looked at this Bicep deployment.  
Now I understood the concept. 
I have removed the deployment.json from this PR.

File: docs/optimization-workbook/workbook/Optimization Workbook.workbook
[the remark this answers is not in the document]
Comment (flanakin) [reply docs/optimization-workbook/workbook/Optimization Workbook.workbook:1]: Don't forget to address this one so we can close the PR.

**Review by flanakin — CHANGES_REQUESTED** [review]
Still need to rename the workbook so the build script works.

**Comment by arthurclares** [comment]
> Still need to rename the workbook so the build script works.

I've renamed the file.
I also updated the workbook.json at /src/workbooks/workbook.json with the same code as /docs/optimization-workbook/ workbook.json

File: docs/optimization-workbook/workbook/workbook.json
Comment (flanakin) [inline docs/optimization-workbook/workbook/workbook.json:1]: Did you mean to put the workbook JSON in docs? We shouldn't duplicate it.

File: docs/optimization-workbook/README.md [hunk 1]
```diff
@@ -10,4 +10,19 @@
...
+
+### Deployment
+* Consider the following least-privilege (minimum) RBAC roles when deploying the workbook:
+
+  * **Reader** and **Workbook Contributor** : allows you to import the workbook, view all of the workbook tabs and save the workbook in a resource group.
+  * **Reader** : allows you to import the workbook and view all of the workbook tabs.
+
+Use the following link to deploy the Optimization Workbook:
+
+   <a href="https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2Farthurclares%2FWAFOptimizationFactory%2Fmain%2FOptmizationFactoryWorkbook%2Fazuredeploy.json" target="_blank"><img src="https://aka.ms/deploytoazurebutton"/></a>
```
Comment (flanakin) [inline docs/optimization-workbook/README.md:27]: I don't want to link out to an external repo given it'll be different than what we have here. Let's remove this for now until we get it deployed to the Azure Quickstart Templates repo. I can do that as soon as this PR is closed.

**Review by flanakin — CHANGES_REQUESTED** [review]

**Review by flanakin — APPROVED** [review]

**Pull request merged**